Gems/AudioSystem/Code/Source/Editor/AudioControlsWriter.cpp

```diff
@@ -322,24 +385,24 @@ namespace AudioControls
     }
 
     //-------------------------------------------------------------------------------------------//
-    void CAudioControlsWriter::CheckOutFile(const AZStd::string& filepath)
+    void CAudioControlsWriter::CheckOutFile(const AZStd::string_view filepath)
     {
         IEditor* editor = GetIEditor();
         IFileUtil* fileUtil = editor ? editor->GetFileUtil() : nullptr;
         if (fileUtil)
         {
-            fileUtil->CheckoutFile(filepath.c_str(), nullptr);
+            fileUtil->CheckoutFile(filepath.data(), nullptr);
         }
     }
 
     //-------------------------------------------------------------------------------------------//
-    void CAudioControlsWriter::DeleteLibraryFile(const AZStd::string& filepath)
+    void CAudioControlsWriter::DeleteLibraryFile(const AZStd::string_view filepath)
     {
         IEditor* editor = GetIEditor();
         IFileUtil* fileUtil = editor ? editor->GetFileUtil() : nullptr;
         if (fileUtil)
         {
-            fileUtil->DeleteFromSourceControl(filepath.c_str(), nullptr);
+            fileUtil->DeleteFromSourceControl(filepath.data(), nullptr);
```

Comment:
Since string_views aren't guaranteed to be null-terminated, instead of invoking `filepath.data()`, this could construct a temporary `AZ::IO::FixedMaxPathString`, which doesn't require any memory allocation and copies to a temporary `AZStd::fixed_string<>`

For example
````
fileUtil->DeleteFromSourceControl(AZ::IO::FixedMaxPathString(filepath).c_str(), nullptr)

Reply:
Will do.